### DISCUSSION TITLE
Best solution for getting the word count?

### DISCUSSION THREAD

**Discussion by shustudios** [ueberdosis/tiptap#1771] [General Questions] [answered]
Hi, I'm looking for an optimized way for getting the number of words in the editor. The problem is that text between nodes (multiple paragraphs) get "joined" when converting the editor's contents into plain text. Right now the only solution i see is to recursively loop through all the nodes to get the text.

Anyone have a better idea? Thanks!

**Comment by sereneinserenade** [comment, marked as answer]
I don't know if this is the best solution, but here's what I've tried.

https://codesandbox.io/s/reverent-babycat-g4yuk

**Reply by shustudios** [reply]
Hi,
Your solution has the same problem as mine - if you write:

this
is
a
test

You wil see it only counts as 1 word. This is because when you read as innertext, the paragraphs gets joined without any spaces between them.

**Reply by shustudios** [reply]
Will probably have to look at using a combination of the existing solution, plus counting all the paragraphs, but I was hoping to find a more efficient method.

**Reply by sereneinserenade** [reply]
Hey there @ironshu  , I did some adjustments and it works fine for me. https://codesandbox.io/s/reverent-babycat-g4yuk

<img width="511" alt="image" src="https://user-images.githubusercontent.com/45892659/130468095-54dededb-fb8d-4806-b9b1-e36ee81879d2.png">

```js
    countTheWords() {
      const proseMirror = document.querySelector(".ProseMirror");

      const innerText = proseMirror.innerText
        .replaceAll("\n", " ")
        .split(" ")
        .filter((s) => s !== "" && s !== " ");

      this.length = innerText.length;
    },


```

**Reply by shustudios** [reply]
hmmm, that didn't work for me, but it was probably because I was using the editor `textContent` property... I'll try your approach. Still wondering if it is efficient, but it is a nice and simple solution 👍

**Reply by shustudios** [reply]
Thanks! Also I found that the issue for me was not the source string, but the regex I was using the split the words. So instead of parsing the html output, seems you can use `editor.state.doc.textContent`  - not sure which is faster though

Edit - scratch that - had my variables mixed up... need to parse the dom as per your solution..
